src/app/atoms/textarea/che-textarea.component.spec.ts

```diff
@@ -0,0 +1,588 @@
+                let divGroup = fixture.debugElement.query(By.css('.group'))!;
+
+                // Act
+                expect(textarea.nativeElement.classList.contains('ng-touched'))
+                    .toBe(false, 'should to be as untouched');
+
+                textarea.nativeElement.value = 'a value';
+                textarea.nativeElement.dispatchEvent(new Event('input'));
+                fixture.detectChanges();
+
+                // Assert
+                expect(divGroup.nativeElement.classList.contains('che-danger'))
+                    .toBe(false, 'should have not the class che-danger');
+
+                // Act
+                textarea.triggerEventHandler('blur', null);
+                fixture.detectChanges();
+
+                // Assert
+                expect(divGroup.nativeElement.classList.contains('che-danger'))
+                    .toBe(false, 'should have not the class che-danger');
+
+            });
+
+
+        });
+        describe('If the textarea is not required', () => {
+            beforeEach(async(() => {
+                // Arrange
+                fixture.destroy();
+                TestBed.resetTestingModule();
+                fixture = createComponent(CheTextareaF2Component);
+                fixture.detectChanges();
+            }));
+            // test 16
+            it('Should  be not invalid to make click or if have a value', () => {
+                // Arrange
+                let textarea = fixture.debugElement.query(By.css('textarea'))!;
+                let divGroup = fixture.debugElement.query(By.css('.group'))!;
+
+                textarea.triggerEventHandler('focus', null);
+                fixture.detectChanges();
+
+                // Assert
+                expect(divGroup.nativeElement.classList.contains('che-danger'))
+                    .toBe(false, 'should have not the class che-danger');
+
+                textarea.nativeElement.value = 'a value';
+                textarea.nativeElement.dispatchEvent(new Event('input'));
+                fixture.detectChanges();
+
+                // Assert
+                expect(divGroup.nativeElement.classList.contains('che-danger'))
+                    .toBe(false, 'should have not the class che-danger');
+            });
+
+        });
+    });
+
+    describe('If the design is by default or border', () => {
+        beforeEach(async(() => {
+            // Arrange
+            fixture.destroy();
+            TestBed.resetTestingModule();
+            fixture = createComponent(CheTextareaF5Component);
+            component = fixture.componentInstance;
+            fixture.detectChanges();
+        }));
+        // test 17
+        it('Should be design border', () => {
+            // Arrange
+            let textarea = fixture.debugElement.query(By.css('textarea'))!;
+            // Act
+
+            // Assert
+            expect(textarea.nativeElement.design).toBeUndefined('Should not to have a design');
```

Comment:
Perhaps it should evaluate that the design value is equal to 'border' in the assertion.